src/unix/notbsd/linux/mod.rs

```diff
@@ -185,6 +185,44 @@ s! {
     }
 }
 
+f! {
+    // FIXME(peterhj, 20160204): should not have pub visibility.
+    pub fn CPUELT(cpu: usize) -> usize {
+      cpu / CPUSET_ULONG_SIZE
+    }
+
+    // FIXME(peterhj, 20160204): should not have pub visibility.
+    pub fn CPUMASK(cpu: usize) -> ::c_ulong {
+      1 << (cpu % CPUSET_ULONG_SIZE)
+    }
+
+    pub fn CPU_ZERO(cpuset: &mut cpu_set_t) -> () {
+      for idx in 0 .. CPU_SETSIZE as usize / CPUSET_ULONG_SIZE {
+        cpuset.bits[idx] = 0;
+      }
+    }
+
+    pub fn CPU_SET(cpu: usize, cpuset: &mut cpu_set_t) -> () {
+      cpuset.bits[CPUELT(cpu)] |= CPUMASK(cpu);
+      ()
+    }
+
+    pub fn CPU_ISSET(cpu: usize, cpuset: &cpu_set_t) -> bool {
+      (cpuset.bits[CPUELT(cpu)] & CPUMASK(cpu)) != 0
+    }
+
+    pub fn CPU_EQUAL(set1: &cpu_set_t, set2: &cpu_set_t) -> bool {
+      set1.bits == set2.bits
+    }
+}
+
+// FIXME(peterhj, 20160204): duplicated from unix::notbsd.
+// intentionally not public, only used for fd_set
+#[cfg(target_pointer_width = "32")]
+const CPUSET_ULONG_SIZE: usize = 32;
+#[cfg(target_pointer_width = "64")]
+const CPUSET_ULONG_SIZE: usize = 64;
```

Comment:
If these constants aren't need for anything but the macros, could they be moved in as just `mem::size_of::<::ulong_t>`?